diff --git a/pycolmap/CMakeLists.txt b/pycolmap/CMakeLists.txt
index da6623b77..407e15970 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -14,8 +14,8 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     add_definitions("-DNOMINMAX")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
     # Enable object level parallel builds in Visual Studio.
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
+    # set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP")
+    # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
 endif()
 
 find_package(colmap REQUIRED)
