diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b45a9f..004ee8b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,4 +37,8 @@ if(MSVC)
 # it crashes with LNK2019 because of that
 # https://github.com/colmap/colmap/blob/9f3a75ae9c72188244f2403eb085e51ecf4397a8/cmake/CMakeConfig.cmake.in#L165
 target_link_libraries(pycolmap PRIVATE ${Boost_FILESYSTEM_LIBRARY} ${Boost_PROGRAM_OPTIONS_LIBRARY} ${Boost_SYSTEM_LIBRARY})
-endif()
\ No newline at end of file
+endif()
+
+if (${COLMAP_CUDA_ENABLED})
+  target_link_libraries(pycolmap PRIVATE CUDA::cudart)
+endif()
